src/go/rdctl/cmd/listSettings.go

```diff
@@ -17,17 +17,22 @@ limitations under the License.
 package cmd
 
 import (
-	"github.com/spf13/cobra"
+  "fmt"
+  "github.com/spf13/cobra"
 )
 
 // listSettingsCmd represents the listSettings command
 var listSettingsCmd = &cobra.Command{
 	Use:   "list-settings",
 	Short: "Lists the current settings.",
-	Long: `Lists the current settings in JSON format.`,
-  RunE: func(cmd *cobra.Command, args []string) error {
-    return doRequest("GET", "list-settings")
-  },
+	Long:  `Lists the current settings in JSON format.`,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		result, err := doRequest("GET", "list-settings")
+    if len(result) > 0 {
```

Comment:
How can `result` be empty, but `err` is also `nil`? That should not be possible.

Reply:
Yeah, I would expect so. But if the server returns a status of 200, you have to assume whatever you get back is the correct data.

Reply:
I guess we should formally define what each API returns in the v1 API spec. I still think that `list-settings` not returning any data is a failure, and should never arrive with a `200` status code.

Reply:
I was getting too fancy here, knowing that if an error was returned, the result would be empty. But it's more idiomatic to just do:
```
    result, err := doSomething
    if err != nil {
      return err
    }
    fmt.Println(string(result))
    return nil
```